### ISSUE TITLE
Passwords are stored in a plaintext format

### ISSUE THREAD

**Issue by flfy** [JavidPack/HEROsMod#24]
I find this to be a huge security flaw. There are MUCH better ways to do this.

Example: hash the passwords with bcrypt and a few salt rounds, which is a pretty secure way of preventing the takeover of user accounts.

I hope this gets addressed soon because its a huge flaw in what I've seen to be an amazing mod.

**Comment by flfy** [comment]
Are you going to sit here and tell me that hashing a password is less safe than storing it in plaintext format?

**Comment by JavidPack** [comment]
You mentioned preventing takeover of accounts, in that case yes. Hashing would do nothing because the server host can edit the json file however they want to take over your account, even if hashing were used. You already trust the server host by joining their server. Other clients don't have access to the database, so they aren't a threat.

I guess if we trust that people are using the "only download signed mods" option in tmodloader, we can hash on the client to prevent issues with password reuse issues and trust that the .tmod itself hasn't been tampered with. Problem with that is I believe that feature is broken on mac and linux right now, unfortunately.

Once 0.11 is out I can begin working on a revamp of the sign in feature. I'll probably just get rid of it in some way.

**Comment by JavidPack** [comment]
Yeah, I know, just gotta get 0.11 tmod released first and then I'll be able to tackle all these heros mod issues
